File: app/javascript/channels/room_channel.js
import consumer from "./consumer"

consumer.subscriptions.create({
  channel: "RoomChannel",
  room_id: document.querySelector('head').dataset.currentRoomId
}, {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    console.log(data);
    if (data['notify'] === 'new-issue') {
      window.app.getIssues();
    } else {
      window.app.getIssues();
      // alert(data['message']);
    }
  },

  speak: function(message) {
    return this.perform('speak', { message: message });
  }
});

File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel
  def subscribed
    stream_from 'room_channel'
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak
    ActionCable.server.broadcast 'room_channel', message: data['message']
  end

  class << self
    def send_message(current_issue)
      body = current_issue.estimates.map do |m|
        "#{m.user.name}: #{m.point}\n"
      end
      ActionCable.server.broadcast(
        'room_channel',
        message: body
      )
    end
  end
end
